sympy/polys/polytools.py

```diff
@@ -7003,7 +6999,7 @@ def real_roots(f, multiple=True, radicals=True):
     .. [1] https://en.wikipedia.org/wiki/Casus_irreducibilis
     """
     try:
-        F = Poly(f, greedy=False)
+        F = Poly(f, extension=True)
```

Comment:
I think that there needs to be some way to control this when calling `real_roots` or `all_roots`. It might make sense for `extension=True` to be the default for these functions but then there at least needs to be a way to pass `extension=False` when calling the functions.

Reply:
Perhaps by adding a new argument? Currently, `real_roots()` or `all_roots()` take arguments `multiple` and `radical`. Or perhaps using `**kwargs`?

Reply:
The obvious choice would be for them to take an argument `extension` that defaults to False. That would be in keeping with other Poly functions.

Reply:
Sure sounds good